lib/src/native/realm_core.dart

```diff
@@ -1039,31 +1039,35 @@ class _RealmCore {
           }
         });
 
-        responseRef.custom_status_code = _CustomErrorCode.noError.code;
+        responseRef.custom_status_code = CustomErrorCode.noError.code;
       } on SocketException catch (_) {
         // TODO: A Timeout causes a socket exception, but not all socket exceptions are due to timeouts
-        responseRef.custom_status_code = _CustomErrorCode.timeout.code;
+        responseRef.custom_status_code = CustomErrorCode.timeout.code;
       } on HttpException catch (_) {
-        responseRef.custom_status_code = _CustomErrorCode.unknownHttp.code;
+        responseRef.custom_status_code = CustomErrorCode.unknownHttp.code;
       } catch (_) {
-        responseRef.custom_status_code = _CustomErrorCode.unknown.code;
+        responseRef.custom_status_code = CustomErrorCode.unknown.code;
       } finally {
         _realmLib.realm_http_transport_complete_request(request_context, response_pointer);
       }
     });
   }
 
   static void logCallback(Pointer<Void> userdata, int levelAsInt, Pointer<Int8> message) {
+    final logger = Realm.logger;
+    if (logger == null) {
+      return;
+    }
```

Comment:
We should always have a logger - if someone doesn't want to log stuff, they can set the level to `Off`.

Reply:
Logging should be done only if requested. I fully disagree that a library should default to logging and explicitly ask for disabling it. It should be the opposite. Relates to your general statement above. If people have issues only then they should enable logging. We should be optimizing the defaults for the general case and it is sync should be working.

Reply:
How would they know they have issues if they don't enable logging? E.g. if they experience a client reset? Or if their writes are reverted due to some permission rules. We enable logging by default in other SDKs and no one has complained about it.

Reply:
Client reset is an error. Any reverted write due to permissions should be an error. 
In my view every logging by default should be illegal. 
Seriously apps should not be logging especially on mobile. Libraries should not be logging by default. Imagine the great user experience integrating five different libraries in your app, which log by default to the console. You can not use it by default to develop anything since it will be a nightmare, so you need to disable all of them. Thus if we log be default we are only choosing a poor default and introduce another integration step.  Logging should be enabled only if explicitly requested to debug something. This is why `print` statements are flagged in Flutter apps. 
To me the other SDKs have chosen a poor default, in anticipation of errors that don't happen at all or should happen as exceptions. 

Remember **"For most users the default settings are the only settings"** 
So If we enable logging by default this is what they are shipping, making people devices slower and draining their batteries. Totally unnecessary.

Reply:
Exactly. **"For most users the default settings are the only settings"** - Hence we should log by default. At least in DEBUG mode.

Reply:
So what if users haven't specified a client reset handler? So the error goes in the void and they get no feedback that something went wrong. Generally, people want to have feedback that something is happening, especially in the early phases of development. Emitting logs is fairly standard and many frameworks and libraries do it. Both on Android and iOS, there are logs about UI misconfiguration, slow rendering, as well as http requests. I don't think it'll be unusual to have Realm log stuff for the users. Again, no one has ever complained about it and the overhead if info logging should be negligible. Unless you have profiled it and indeed it has measurable impact on CPU usage/battery life, I would strongly advise we stick to established defaults. We can always change them later, but that should be brought up for a broader discussion rather than have different SDKs behave differently.

Reply:
For the record: I strongly disagree with the decision to leave a default logger with info level even in Release.